src/arrayMethodSort.js

```diff
@@ -5,7 +5,48 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let compare = compareFunction;
+
+    // compare function default - comparing as strings
+    if (compare === undefined
+      || typeof compare !== 'function') {
+      compare = (a, b) => {
+        const _a = String(a);
+        const _b = String(b);
+
+        // not sure why localCompare would not work for cyryllic letters
+        if (_a > _b) {
+          return 1;
+        }
+
+        if (_b > _a) {
+          return -1;
+        }
+
+        return 0;
+      };
+    }
+
+    // variable for do while loop
+    let sorting;
```

Comment:
the only change I would do is to rename `let sorting` to `let isSorted`. Basically, for booleans we use prefixes `is` or `has` and then a verb in past tense. reade more [here](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)